### PR TITLE
Add new `Lint/DeprecatedConstants` cop

### PR DESCRIPTION
**Pull request by koic**
This PR adds new `Lint/DeprecatedConstants` cop.

This cop checks for deprecated constants.

It has `DeprecatedConstants` option. If there is an alternative method, you can set the value as a string. Or set `null` to the value if there is no alternative.
By default, `NIL`, `TRUE`, `FALSE` and `Random::DEFAULT` are configured.

```ruby
# bad
NIL
TRUE
FALSE
Random::DEFAULT # Return value of Ruby 2 is `Random` instance, Ruby 3.0 is `Random` class.

# good
nil
true
false
Random.new # `::DEFAULT` has been deprecated in Ruby 3, `.new` is compatible with Ruby 2.
```

The following are these sources of information.

- Remove (deprecated) `NIL`, `TRUE`, and `FALSE` ... https://github.com/ruby/ruby/commit/62554ca
- Deprecate `Random::DEFAULT` ... https://bugs.ruby-lang.org/issues/17351

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
Nice cop, but it might also be a good idea to add some data about the Ruby version where something got deprecated to the config as well. This will allow us to have more informative messages.

**Comment by koic** [comment]
That's good idea! I added deprecated version for deprecated constants.

File: lib/rubocop/cop/lint/deprecated_constants.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        def consntant_name(node, nested_constant_name)
+          return nested_constant_name unless node.namespace.const_type?
+
+          consntant_name(node.namespace, "#{node.namespace.short_name}::#{nested_constant_name}")
+        end
+
+        def message(good, bad, deprecated_version)
+          if deprecated_version
+            deprecated_message = ", which is deprecated in Ruby #{deprecated_version}"
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/deprecated_constants.rb:60]: "deprecated since" would be a slightly better wording here

Comment (koic) [reply lib/rubocop/cop/lint/deprecated_constants.rb]: I see! Thank your for the suggestion!

File: config/default.yml [hunk 2]
```diff
@@ -1469,6 +1469,25 @@ Lint/DeprecatedClassMethods:
...
+  # Or set `null` to the value if there is unknown deprecated version.
+  #
+  # DeprecatedConstants:
+  #   'DEPRECATED_CONSTANT': ['alternative_value', 'deprecated_version']
+  #
+  DeprecatedConstants:
+    'NIL': ['nil', '2.4']
+    'TRUE': ['true', '2.4']
+    'FALSE': ['false', '2.4']
+    'Random::DEFAULT': ['Random.new', '3.0']
```
Comment (koic) [inline config/default.yml:1489]: I'm not sure whether array is good. For example, hash may be preferable. Please let me know if you have feedback 😅

Comment (bbatsov) [reply config/default.yml:1489]: I agree that hash would be better. This would spare us from having to use explicit `null`s.

Comment (koic) [reply config/default.yml]: > This would spare us from having to use explicit `null`s.

Sure! That makes sense :-) I updated it!

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**